### PR TITLE
##3699938 Epic Comments

### PR DESCRIPTION
**Pull request by andela-oowonikoko**
### What does this PR do ?

- Create comments for a particular contribution
- Fetch all the comments that belong to a contribution
- Implements deletetion of single comment

### Description of Task to be completed ? 
- Create an endpoint for creating comments by a contribution id  
   `/contributions/:contribution_id/comments`
- Create an endpoint for fetching comments by a contribution id `/contributions/:id/comments`
- Create an endpoint for Deleting a comment by id `Delete` `/comments/:comment_id`
- Write tests for the endpoints

### How should this be manually tested?

- At the root of the project directory run the following command:
``` bash
  pytest -v or pytest
```
### What are the relevant pivotal tracker stories?

##3699938

### CODE REVIEW

**Comment by njirap** [comment]
@andela-oowonikoko my heart is not at peace with the first endpoint `/api/v1/contributions/?include=comments` Is it very necessary? cc @andela-kakpobome

**Comment by andela-oowonikoko** [comment]
@njirap According to the story on pivotal-tracker (https://www.pivotaltracker.com/story/show/150002309) by @andela-kakpobome, this endpoint is necessary to get all contributions with their comments.

**Comment by njirap** [comment]
char_limit


git titles have a character limit.. that's why this title is overflowing into the body and the sentence is not complete. Shorten the title while maintaining it's correctness as a grammatical expression

**Comment by njirap** [comment]
get rid of the endpoint ` /api/v1/contributions/?include=comments` reference the slack discussion we had on the same

**Comment by njirap** [comment]
char

just noticed this after merging it to develop. I will be renaming it to shroten it

**Comment by andela-oowonikoko** [comment]
The slack discussion says we should create the endpoint `contributions/:id/comments` instead of having to include `comments` into the contributions endpoint. And we should also include `comment_count` as part of the response. I'll work on that now.

**Comment by andela-oowonikoko** [comment]
@njirap I have removed the endpoint `/api/v1/contributions/?include=comments` and added `comment_count` to the response.

I have also squashed the commits and renamed it

File: api/views/contribution_comments.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+from uuid import uuid4
```
Comment (njirap) [inline api/views/contribution_comments.py]: I am moving all this into the contributions views

File: api/views/contributions.py [hunk 2]
```diff
@@ -34,6 +41,8 @@ def get(self):
 
         for c in _contributions:
             contrib = c.serialize()
+            contrib['commentCount'] = len(list(map(serialize_thread_comments,
+                                             c.comments)))
```
Comment (njirap) [inline api/views/contributions.py]: what's wrong with `len(c.comments)`?

File: api/models.py [hunk 3]
```diff
@@ -50,6 +49,23 @@ class Tag(db.Model, SerializerMixin):
...
+    """
+    Define fields for comments
+    """
+    __tablename__ = "Comment"
+    id = db.Column(db.String(),primary_key=True)
+    user_id = db.Column(db.String(),nullable=False)
+    contribution_id = db.Column('contribution_id', db.String,
+                                     db.ForeignKey('Contribution.id'))
+    message = db.Column(db.String(), nullable=False)
+    parent_id = db.Column(db.String(), db.ForeignKey('Comment.id'))
```
Comment (njirap) [inline api/models.py]: whatt is this field for?

File: api/models.py [hunk 4]
```diff
@@ -50,6 +49,23 @@ class Tag(db.Model, SerializerMixin):
...
+    created_at = db.Column(db.DateTime, default=datetime.utcnow)
+    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
+    parent = db.relationship('Comment', backref='replies', remote_side='Comment.id')
+
+    def __repr__(self):
+         return '<Comment %r>' % self.message
```
Comment (njirap) [inline api/models.py]: 2 blank spaces instead of 1

File: api/models.py [hunk 5]
```diff
@@ -50,6 +49,23 @@ class Tag(db.Model, SerializerMixin):
     def __repr__(self):
         return '<Tag %r>' % self.name
```
Comment (njirap) [inline api/models.py]: 2 blank spaces instead of 1

File: api/views/contributions.py (hunk 2 above)
Comment (andela-oowonikoko) [inline api/views/contributions.py]: Fixed. I tried `len(c.comments)` and I was getting an error so I researched further into the error and all I had to do was `len(c.comments.all())`

File: api/models.py (hunk 3 above)
Comment (andela-oowonikoko) [inline api/models.py]: The `parent_id` field is meant to be `null` for regular comments and meant to have an `id` for replies to a comment

File: api/models.py (hunk 4 above)
Comment (andela-oowonikoko) [inline api/models.py]: Fixed 👍

File: api/models.py (hunk 5 above)
Comment (andela-oowonikoko) [inline api/models.py]: Fixed 👍

**Comment by njirap** [comment]
LGTM

**Pull request merged**